numba/unicode.py

```diff
@@ -531,6 +553,36 @@ def find_impl(a, b):
         return find_impl
 
 
+@overload_method(types.UnicodeType, 'count')
+def unicode_count(src, sub, start=None, end=None):
+
+    if not (start is None or isinstance(start, (types.Omitted, types.Integer, types.Optional))):
+        raise TypingError("Start arg must be of type Integer or None")
+
+    if isinstance(sub, types.UnicodeType):
+        def count_impl(src, sub, start=start, end=end):
+            count = 0
+            src_len = len(src)
+            sub_len = len(sub)
+
+            # normalize start and end indexes to define search range
+            start = normalize_slice_arg(start, src_len, 0)
+            end = normalize_slice_arg(end, src_len, src_len)
+
+            if (sub_len == 0):       #special case when substring is empty
+                return _count_special_case(start, end, src_len)
```

Comment:
@stuartarchibald Are you asking about _count_special_case? If yes then it needs to be a function because it has multiple conditions so including that in a function makes the code more readable.

Reply:
It is similar to len(src[slice(start, end)]) + 1 but not exactly like that so I think it will be better to put it like this in a separate function. Python has unique behavior when it comes to using such methods with empty strings and more than one condition is needed to cover all the corner cases related to this special case. But if you have other suggestions regarding this I am open for that.

Reply:
I think that this is because the `src` string isn't pre-sliced so the computation has to happen. Could the function body just be inlined here, it's small and only called from this location?

Reply:
Okay will do that.